### PR TITLE
[DOCU-2823] New 3.1.x Upgrade doc

### PR DESCRIPTION
**Pull request by cloudjumpercat**
Signed-off-by: Diana <75819066+cloudjumpercat@users.noreply.github.com>

### Summary

Created a new upgrade doc for 3.1.x that contains the incremental upgrade scenarios for 3.1.x.
### Reason

DOCU-2823
### Testing

- /gateway/latest/upgrade-31x (is the file name okay for this one?)
- Check the redirect that I added. Is it correct and does it work?

### CODE REVIEW

**Comment by cloudjumpercat** [comment]
@flrgh Hi! Would you be able to review the upgrade steps I added for 3.1 to make sure they are technically accurate? Thanks!

File: src/gateway/upgrade-31x.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+Rolling upgrades from 1.x or 2.x to 3.1.x for hybrid mode are not supported.
+To upgrade to 3.1.x, you must first [upgrade to 3.0.x](/gateway/3.0.x/upgrade).
+
+## Upgrade from 3.0.x to 3.1.x
+
+### Traditional mode
+
+{:.note}
+> These instructions will only work once 3.1.x is available.
```
Comment (acgoldsmith) [inline src/gateway/upgrade-31x.md:216]:
```suggestion

```

File: src/gateway/upgrade-31x.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+{% endnavtabs %}
+
+## Upgrade from 1.x/2.x to 3.1.x
+
+### Traditional mode
+
+Due to the changes in 3.1.x, you must upgrade from 2.8.2 or later to 3.0.x, then 3.1.x.
+
+{:.note}
+> These instructions will only work once 3.1.x is available.
```
Comment (acgoldsmith) [inline src/gateway/upgrade-31x.md:170]:
```suggestion

```

File: src/gateway/upgrade-31x.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+distinction between major, minor, and patch versions.
+
+The upgrade to {{page.kong_version}} is a **minor** upgrade.
+The lowest version that Kong {{page.kong_version}} supports migrating from is 3.0.x.
+
+{:.important}
+> **Important**: Blue-green migration in traditional mode for versions below 2.8.2 to 3.0.x is not supported.
+The upcoming 2.8.2 release will include blue-green migration support. If you want
+to perform migrations for traditional mode with no downtime,
+upgrade to 2.8.2, [then migrate to {{page.kong_version}}](#migrate-db).
```
Comment (acgoldsmith) [inline src/gateway/upgrade-31x.md:27]:
```suggestion
The 2.8.2 release includes blue-green migration support. If you want
to perform migrations for traditional mode with no downtime,
upgrade to 2.8.2, [then migrate to {{page.kong_version}}](#migrate-db).
```

**Review by acgoldsmith — APPROVED** [review]
LGTM - some changes related to release state of 3.1.x and 2.8.2, plus I've reviewed the inclusions from @flrgh that were his comments on PR4849 and they LGTM!

**Pull request merged**